drivers: spi: esp32: fix frequency config

sylvioalves opened the pull request:
Current implementation is using hardcoded frequency
and not the one provided by the spi api. 
This also fix missing CS pin configuration when
custom GPIO is selected.


Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves committed:
drivers: spi: esp32: fix frequency config

Current implementation is using hardcoded frequency
and not the one provided by the spi api.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves commented:
@bbilas PTAL

bbilas commented:
> @bbilas PTAL

Could you move that change related to SPI_CONTEXT_CS_GPIOS_INITIALIZE macro into a separate commit? It seems to be not related to frequency config at all.

sylvioalves committed:
drivers: spi: esp32: fix CS gpio initialization

SPI_CONTEXT_CS_GPIOS_INITIALIZE call was not considering
proper SPI driver instance ID.

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

sylvioalves commented:
Just pushed to retrigger buildkite.
